PWGCF/FemtoDream/FemtoDreamDetaDphiStar.h

```diff
@@ -0,0 +1,198 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file FemtoDreamDetaDphiStar.h
+/// \brief FemtoDreamDetaDphiStar - Checks particles for the close pair rejection.
+/// \author Laura Serksnyte, TU München, laura.serksnyte@tum.de
+
+#ifndef ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMDETADPHISTAR_H_
+#define ANALYSIS_TASKS_PWGCF_FEMTODREAM_FEMTODREAMDETADPHISTAR_H_
+
+#include "PWGCF/DataModel/FemtoDerived.h"
+#include "Framework/HistogramRegistry.h"
+#include <string>
+
+namespace o2::analysis
+{
+namespace femtoDream
+{
+namespace
+{
+static std::array<std::array<std::shared_ptr<TH2>, 2>, 2> histdetadpi{};
+static std::array<std::array<std::shared_ptr<TH2>, 9>, 2> histdetadpiRadii{};
+} // namespace
+
+/// \class FemtoDreamDetaDphiStar
+/// \brief Class to check particles for the close pair rejection.
+/// \tparam partOne Type of particle 1 (Track/V0/Cascade/...)
+/// \tparam partTwo Type of particle 2 (Track/V0/Cascade/...)
+template <o2::aod::femtodreamparticle::ParticleType partOne, o2::aod::femtodreamparticle::ParticleType partTwo>
+class FemtoDreamDetaDphiStar
+{
+ public:
+  /// Destructor
+  virtual ~FemtoDreamDetaDphiStar() = default;
+  /// Initalization of the histograms and setting required values
+  void init(HistogramRegistry* registry, HistogramRegistry* registryQA, float ldeltaPhiMax, float ldeltaEtaMax, float lmagfield, bool lplotForEveryRadii)
+  {
+    deltaPhiMax = ldeltaPhiMax;
+    deltaEtaMax = ldeltaEtaMax;
+    magfield = lmagfield;
+    plotForEveryRadii = lplotForEveryRadii;
+    mHistogramRegistry = registry;
+    mHistogramRegistryQA = registryQA;
+
+    if constexpr (mPartOneType == o2::aod::femtodreamparticle::ParticleType::kTrack && mPartTwoType == o2::aod::femtodreamparticle::ParticleType::kTrack) {
+      histdetadpi[0][0] = std::get<std::shared_ptr<TH2>>(mHistogramRegistry->add((static_cast<std::string>(histNames[0][0])).c_str(), "; #Delta #eta; #Delta #phi", kTH2F, {{100, -2.5, 2.5}, {100, -2.5, 2.5}}));
```

Comment:
hi Laura,
we now have an add() function in the HistogramRegistry that directly returns the histogram pointer (no more need to extract it manually from the HistType variant via std::get):
```cpp
histdetadpi[0][0] = mHistogramRegistry->add<TH2>(...)
```
(see for instance here: https://github.com/AliceO2Group/O2Physics/blob/master/PWGLF/Tasks/spectraTPC.cxx#L71)
Cheers,
  Mario